cmds/boot/boot.go

```diff
@@ -0,0 +1,451 @@
+			}
+		}
+	}
+	return returnValue, nil
+}
+
+// getSupportedFilesystem returns all block file system supported by the linuxboot kernel
+func getSupportedFilesystem() ([]string, error) {
+	var returnValue []string
+	var err error
+	fs, err := ioutil.ReadFile("/proc/filesystems") 
+	if err != nil {
+		return returnValue, err
+	}
+	for _, f := range strings.Split(string(fs), "\n") {
+		n := strings.Fields(f)
+		if len(n) != 1 {
+			continue
+		}
+		returnValue=append(returnValue,n[0])
+	}
+	return returnValue, err
+
+}
+
+// mountEntry tries to mount a specific block device
+func mountEntry(path string, supportedFilesystem []string) (bool,error) {
+	var returnValue bool
+	var err error
+	exist, err := os.Stat("/u-root")
+        if exist == nil {
+                err=syscall.Mkdir("/u-root", 0777)
+                if ( err != nil ) {
+                        return false,err
+                }
+        }
+	var flags uintptr
+	// Was supposed to be unecessary for kernel 4.x.x
+	if verbose {
+		log.Printf("/dev/" + path)
+	}
+	for _, filesystem := range supportedFilesystem {
+		flags = syscall.MS_MGC_VAL
+		// Need to load the filesystem kind supported
+		exist, err = os.Stat("/u-root/"+path)
+                if exist == nil {
+                        err = syscall.Mkdir("/u-root/"+path, 0777)
+                        if ( err != nil ) {
+                                return false,err
+                        }
+                }
+		err := syscall.Mount("/dev/"+path, "/u-root/"+path, filesystem, flags, "")
+		if err == nil {
+			return true, nil
+		}
+	}
+	returnValue = false
+	return returnValue, nil
+}
+
+func umountEntry(path string) bool {
+	var returnValue bool
+	var flags int
+	// Was supposed to be unecessary for kernel 4.x.x
+	flags = syscall.MNT_DETACH
+	err := syscall.Unmount(path, flags)
+	if err == nil {
+		return true
+	}
+	returnValue = false
+	return returnValue
+}
+
+// checkBootEntry is looking for grub.cfg file
+// and return absolute path to it
+func checkBootEntry(mountPoint string) string {
+	_, err := os.Stat(mountPoint + "/boot")
+	if err == nil {
+		// The boot directory is there
+		_, err2 := os.Stat(mountPoint + "/boot" + "/grub")
+		if err2 == nil {
+			_, err3 := os.Stat(mountPoint + "/boot" + "/grub" + "/grub.cfg")
+			if err3 == nil {
+				// found
+				return (mountPoint + "/boot" + "/grub")
+			}
+		}
+	}
+	return ""
+
+}
+
+// getFileMenuContent is parsing a grub.cfg file
+// input: absolute directory path to grub.cfg
+// output: Return a list of strings with the following format
+//	 line[3*x] - menuconfig
+//	 line[3*x+1] - linux kernel + boot options
+// 	 line[3*x+2] - initrd
+// and the default boot entry configured into grub.cfg
+func getFileMenuContent(path string) ([]string, int, error) {
+	var returnValue []string
+	var err error
+	file, err := os.Open(path + "/grub.cfg")
+	if ( err != nil ) {
+		return returnValue,0,err
+	}
+	scanner := bufio.NewScanner(file)
```

Comment:
If this function took a string or []string it would be much easier

Reply:
I am not sure to follow what you meant, you want me to refactor the return value into a []string only ?

Reply:
The work involved in doing a stat is a large fraction of the amount it takes to do the read. So you could ioutil.ReadFile and return
lines []string, name string, error

and that way you'd have the file contents and the path.